node/actors/control_api.go

```diff
@@ -181,62 +180,198 @@ func (api *ControlAPI) subscribe() error {
-	// TODO
+	req := new(nodecontrol.AuctionRequestJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal auction request", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to unmarshal auction request: %s", err))
+		return
+	}
+
+	convertedAgentType := make([]string, len(req.AgentType))
+	for _, at := range req.AgentType {
+		convertedAgentType = append(convertedAgentType, string(at))
+	}
+
+	auctResp, err := api.auctionResponse(string(req.Os), string(req.Arch), convertedAgentType, req.Tags.Tags)
+	if err != nil {
+		api.logger.Error("Failed to generate auction response", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to generate auction response: %s", err))
+		return
+	}
+
+	if auctResp == nil {
+		respondEnvelope(m, AuctionResponseType, 404, nil, "node does not satisfy auction request")
+		return
+	}
+
+	respondEnvelope(m, AuctionResponseType, 200, auctionResponseFromProto(auctResp), "")
 }
 
 func (api *ControlAPI) handleDeploy(m *nats.Msg) {
-	// TODO
+	req := new(nodecontrol.StartWorkloadRequestJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal deploy request", slog.Any("error", err))
+		respondEnvelope(m, RunResponseType, 500, nil, fmt.Sprintf("failed to unmarshal deploy request: %s", err))
+		return
+	}
+
+	ctx := context.Background()
+	_, agent, err := api.self.ActorSystem().ActorOf(ctx, req.WorkloadType)
+	if err != nil {
+		api.logger.Error("Failed to locate agent actor", slog.String("type", req.WorkloadType), slog.Any("error", err))
+		respondEnvelope(m, RunResponseType, 500, nil, fmt.Sprintf("failed to locate [%s] agent actor: %s", req.WorkloadType, err))
+		return
+	}
+
+	askResp, err := api.self.Ask(ctx, agent, startRequestToProto(req))
+	if err != nil {
+		api.logger.Error("Failed to start workload", slog.Any("error", err))
+		respondEnvelope(m, InfoResponseType, 500, nil, fmt.Sprintf("Failed to start workload: %s", err))
+		return
+	}
+
+	protoResp, ok := askResp.(*actorproto.WorkloadStarted)
+	if !ok {
+		api.logger.Error("Start workload response from agent was not the correct type")
+		respondEnvelope(m, RunResponseType, 500, nil, "Agent returned the wrong data type")
+		return
+	}
+
+	respondEnvelope(m, RunResponseType, 200, startResponseFromProto(protoResp), "")
 }
 
-func (api *ControlAPI) handleInfo(m *nats.Msg) {
+func (api *ControlAPI) handleUndeploy(m *nats.Msg) {
+	req := new(nodecontrol.StopWorkloadRequestJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal undeploy request", slog.Any("error", err))
+		respondEnvelope(m, StopResponseType, 500, nil, fmt.Sprintf("failed to unmarshal undeploy request: %s", err))
+		return
+	}
 
-	req := &actorproto.QueryWorkloads{}
+	ctx := context.Background()
+	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
```

Comment:
we should be sending the undeploy request directly to the agent, the same way we send the deploy request. The agent super should only be involved in operations that involve treating the list of agents as an aggregate or starting/stopping agents.

Reply:
Remember that undeploy here means undeploy a workload, not stop the agent.